test/hotspot/jtreg/compiler/vectorapi/VectorGatherMaskFoldingTest.java

```diff
@@ -160,23 +160,23 @@ public class VectorGatherMaskFoldingTest {
     public static void testTwoLongVectorLoadGatherNotEqualArray() {
         LongVector res = LongVector.fromArray(L_SPECIES, longArray, 0, longIndices, 0);
         LongVector res2 = LongVector.fromArray(L_SPECIES, longArray2, 0, longIndices, 0);
-        Asserts.assertNotEquals(res, res2);
+        Asserts.assertFalse(L_SPECIES.length() != 1 && res.equals(res2));
```

Comment:
I think you don't need this change (`longArray` and `longArray2` are different even with length 1).